lib/eu_central_bank.rb

```diff
@@ -155,58 +148,54 @@ def import_rates(format, s, opts = {})
 
   def check_currency_available(currency)
     currency_string = currency.to_s
-    return true if currency_string == "EUR"
+
+    return true if currency_string == 'EUR'
     return true if CURRENCIES.include?(currency_string)
-    raise CurrencyUnavailable, "No rates available for #{currency_string}"
+
+    raise Errors::CurrencyUnavailable, "No rates available for #{currency_string}"
   end
 
   protected
 
-  def doc(cache, url=ECB_RATES_URL)
-    rates_source = !!cache ? cache : url
-    Nokogiri::XML(open(rates_source)).tap { |doc| doc.xpath('gesmes:Envelope/xmlns:Cube/xmlns:Cube//xmlns:Cube') }
-  rescue Nokogiri::XML::XPath::SyntaxError
-    Nokogiri::XML(open(url))
+  def update_store(exchange_rate, date)
+    store.add_rate(
+      exchange_rate.fetch(:from),
+      exchange_rate.fetch(:to),
+      BigDecimal(exchange_rate.fetch(:rate)),
+      date
+    )
+  end
+
+  def parse_import(content)
+    JSON.parse(content, symbolize_names: true)
+  rescue JSON::ParserError
+    begin
+      YAML.load(content) # rubocop:disable Security/YAMLLoad
```

Comment:
Why not use `safe_load`, instead?

Reply:
because it contains date strings and fails:  https://github.com/ruby/psych/issues/262